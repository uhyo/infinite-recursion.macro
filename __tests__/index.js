const pluginTester = require("babel-plugin-tester").default;
const plugin = require("babel-plugin-macros");

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename },
  tests: [
    `
      import rec from '../../infinite-recursion.macro'

      console.log(rec(function sumTo(num){
        return num <= 0 ? 0 : num + sumTo(num - 1);
      }, 1e7));
    `,
  ],
});
