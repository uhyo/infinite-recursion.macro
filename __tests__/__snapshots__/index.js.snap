// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros basic: basic 1`] = `

import rec from '../../infinite-recursion.macro'

console.log(rec(function sumTo(num){
  return num <= 0 ? 0 : num + sumTo(num - 1);
}, 1e7));

      ↓ ↓ ↓ ↓ ↓ ↓

import { runRecursive as _runRecursive } from "../../infinite-recursion.macro/lib/runtime";
console.log(
  _runRecursive(function* sumTo(num) {
    return num <= 0 ? 0 : num + (yield [num - 1]);
  }, 1e7)
);


`;

exports[`macros bothNamedAndDefault: bothNamedAndDefault 1`] = `

import rec1, { rec as rec2 } from '../../infinite-recursion.macro'

console.log(rec1(function sumTo(num){
  return num <= 0 ? 0 : num + sumTo(num - 1);
}, 1e7));
console.log(rec2(function fib(num){
  return num <= 1 ? num : fib(num - 1) + fib(num - 2)
}, 1e7));

      ↓ ↓ ↓ ↓ ↓ ↓

import { runRecursive as _runRecursive } from "../../infinite-recursion.macro/lib/runtime";
console.log(
  _runRecursive(function* sumTo(num) {
    return num <= 0 ? 0 : num + (yield [num - 1]);
  }, 1e7)
);
console.log(
  _runRecursive(function* fib(num) {
    return num <= 1 ? num : (yield [num - 1]) + (yield [num - 2]);
  }, 1e7)
);


`;

exports[`macros multipleRecs: multipleRecs 1`] = `

import rec from '../../infinite-recursion.macro'

console.log(rec(function sumTo(num){
  return num <= 0 ? 0 : num + sumTo(num - 1);
}, 1e7));
console.log(rec(function fib(num){
  return num <= 1 ? num : fib(num - 1) + fib(num - 2)
}, 1e7));

      ↓ ↓ ↓ ↓ ↓ ↓

import { runRecursive as _runRecursive } from "../../infinite-recursion.macro/lib/runtime";
console.log(
  _runRecursive(function* sumTo(num) {
    return num <= 0 ? 0 : num + (yield [num - 1]);
  }, 1e7)
);
console.log(
  _runRecursive(function* fib(num) {
    return num <= 1 ? num : (yield [num - 1]) + (yield [num - 2]);
  }, 1e7)
);


`;

exports[`macros multipleReferencesInOneRec: multipleReferencesInOneRec 1`] = `

import rec from '../../infinite-recursion.macro'

console.log(rec(function fib(num){
  return num <= 1 ? num : fib(num - 1) + fib(num - 2)
}, 1e7));

      ↓ ↓ ↓ ↓ ↓ ↓

import { runRecursive as _runRecursive } from "../../infinite-recursion.macro/lib/runtime";
console.log(
  _runRecursive(function* fib(num) {
    return num <= 1 ? num : (yield [num - 1]) + (yield [num - 2]);
  }, 1e7)
);


`;

exports[`macros named: named 1`] = `

import { rec } from '../../infinite-recursion.macro'

console.log(rec(function sumTo(num){
  return num <= 0 ? 0 : num + sumTo(num - 1);
}, 1e7));

      ↓ ↓ ↓ ↓ ↓ ↓

import { runRecursive as _runRecursive } from "../../infinite-recursion.macro/lib/runtime";
console.log(
  _runRecursive(function* sumTo(num) {
    return num <= 0 ? 0 : num + (yield [num - 1]);
  }, 1e7)
);


`;

exports[`macros nested: nested 1`] = `

import { rec } from '../../infinite-recursion.macro'

console.log(rec(function sumTo(num){
  const f = rec(function fib(num){
    return num <= 1 ? num : fib(num - 1) + fib(num - 2)
  }, 1e5);

  return num <= 0 ? 0 : num + sumTo(num - 1);
}, 1e7));

      ↓ ↓ ↓ ↓ ↓ ↓

import { runRecursive as _runRecursive } from "../../infinite-recursion.macro/lib/runtime";
console.log(
  _runRecursive(function* sumTo(num) {
    const f = _runRecursive(function* fib(num) {
      return num <= 1 ? num : (yield [num - 1]) + (yield [num - 2]);
    }, 1e5);

    return num <= 0 ? 0 : num + (yield [num - 1]);
  }, 1e7)
);


`;
